File: rc-application/src/main/java/com/transkript/reportcard/business/service/i/SchoolService.java
package com.transkript.reportcard.business.service.i;

import com.transkript.reportcard.api.dto.SchoolDto;
import com.transkript.reportcard.api.dto.response.EntityResponse;
import com.transkript.reportcard.data.entity.School;
import java.util.List;

public interface SchoolService {
    
    School getEntity( Long var1);
    
    EntityResponse create( SchoolDto var1);
    
    List<SchoolDto> getAllDto();
    
    SchoolDto getDto( Long var1);
    
    EntityResponse update( SchoolDto schoolDto);
    
    void delete(Long var1);
}

File: rc-application/src/main/java/com/transkript/reportcard/business/service/i/SchoolService.kt
package com.transkript.reportcard.business.service.i

import com.transkript.reportcard.api.dto.SchoolDto
import com.transkript.reportcard.api.dto.response.EntityResponse
import com.transkript.reportcard.data.entity.School

interface SchoolService {
    fun getEntity(id: Long?): School?
    fun create(schoolDto: SchoolDto?): EntityResponse?
    val schools: List<SchoolDto?>?
    fun getDto(id: Long?): SchoolDto?
    fun update(id: Long?, schoolDto: SchoolDto?): EntityResponse?
    fun delete(id: Long?): String?
}
